src/shared/exec-directory-util.c

```diff
@@ -0,0 +1,15 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "exec-directory-util.h"
+#include "string-table.h"
+
+/* This table maps ExecDirectoryType to the setting it is configured with in the unit */
+static const char* const exec_directory_type_table[_EXEC_DIRECTORY_TYPE_MAX] = {
```

Comment:
Shouldn't this reside in unit-def.[ch]? Seems unnecessary to add a new source file for trivial stuff like this?

Reply:
that means the code is pulled in the public library too though - having it in shared/ means it's not, and @keszybz usually cares a lot about code size

Reply:
Actually, this is more complicated. A function that is in the shared library, either directly or pulled in via one of the static library that are wholly linked into the shared library, is present in one copy, no matter if used or unused, period. A function that is in the static library is copied separately into every place that uses it. But since most of our top-level code links to libshared, it doesn't really mater if the code is defined in src/shared or src/basic, we'll end up with it in libsystemd-shared either way.